Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -943,6 +943,79 @@ public AndConstraint<TAssertions> ContainInOrder(params T[] expected)
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Expects the current collection to contain the specified elements in the exact same order, and to be consecutive.
+    /// using their <see cref="object.Equals(object)" /> implementation.
+    /// </summary>
+    /// <param name="expected">An <see cref="IEnumerable{T}"/> with the expected elements.</param>
+    public AndConstraint<TAssertions> ContainInConsecutiveOrder(params T[] expected)
+    {
+        return ContainInConsecutiveOrder(expected, string.Empty);
+    }
+
+    /// <summary>
+    /// Expects the current collection to contain the specified elements in the exact same order, and to be consecutive.
+    /// </summary>
+    /// <remarks>
+    /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+    /// </remarks>
+    /// <param name="expected">An <see cref="IEnumerable{T}"/> with the expected elements.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="expected"/> is <c>null</c>.</exception>
+    public AndConstraint<TAssertions> ContainInConsecutiveOrder(IEnumerable<T> expected, string because = "",
+        params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot verify ordered containment against a <null> collection.");
+
+        bool success = Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is not null)
+            .FailWith("Expected {context:collection} to contain {0} in order{reason}, but found <null>.", expected);
+
+        if (success)
+        {
+            IList<T> expectedItems = expected.ConvertOrCastToList();
+            IList<T> actualItems = Subject.ConvertOrCastToList();
+
+            int subjectIndex = 0;
+
+            Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
+            int index;
+            int highestIndex = 0;
+            for (index = 0; index < expectedItems.Count; index++)
+            {
+                T expectedItem = expectedItems[index];
+                int previousSubjectIndex = subjectIndex;
+                subjectIndex = IndexOf(actualItems, expectedItem, subjectIndex, areSameOrEqual);
+                highestIndex = Math.Max(index, highestIndex);
```

Comment:
🤔 What do you need the highest index for? Why not just use the `index` in the failure message?

Reply:
The test on line 76 in the test class exercises this. 
If we didn't had `highestIndex` and used `index` the error message would contain "3 (index 0)" instead of the current "3 (index 2)". 
I think it can be described as the highest index of a found item that was not consecutive.

Reply:
That's it exactly, when just looking for items in order we only search once, and so know what the highest index is. When searching for items in consecutive order we can search multiple times so need to keep track of the best match we can make